torch/onnx/symbolic_helper.py

```diff
@@ -667,8 +667,15 @@ def _select_helper(g: jit_utils.GraphContext, self, dim, index, apply_reshape=Tr
                 g, index, g.op("Constant", value_t=torch.LongTensor([1]))
             )
 
-    index_scalar_type = index.type().scalarType()
-    if index_scalar_type is None or index_scalar_type not in {"Long", "Int"}:
+    try:
+        index_scalar_type = _type_utils.JitScalarType.from_value(index)
+    except errors.OnnxExporterError:
+        index_scalar_type = None
+
+    if index_scalar_type is None or index_scalar_type.scalar_name() not in {
+        "Long",
+        "Int",
+    }:
```

Comment:
Doesn't have to be this PR - I can see the value in using the enum instead of strings now that we have `index_scalar_type` being the enum

Reply:
done